lib/src/arithmetic/addend_compressor.dart

```diff
@@ -0,0 +1,253 @@
+    sum <= compressBits.xor();
+    carry <=
+        mux(compressBits[0], compressBits.slice(2, 1).or(),
+            compressBits.slice(2, 1).and());
+  }
+}
+
+/// Compress terms
+enum _CompressTermType {
+  /// A carry term
+  carry,
+
+  /// A sum term
+  sum,
+
+  /// A partial product term (from the original matrix)
+  pp
+}
+
+/// A compression term
+class _CompressTerm implements Comparable<_CompressTerm> {
+  /// The type of term we have
+  late final _CompressTermType type;
+
+  /// The inputs that drove this Term
+  late final List<_CompressTerm> inputs;
+
+  /// The row of the terminal
+  final int row;
+
+  /// The column of the term
+  final int col;
+
+  /// The Logic wire of the term
+  final Logic logic;
+
+  /// Estimated delay of the output of this CompessTerm
+  late double delay;
+
+  /// Estimated delay of a Sum term
+  static const sumDelay = 1.0;
+
+  /// Estimated delay of a Carry term
+  static const carryDelay = 0.75;
+
+  /// CompressTerm constructor
+  _CompressTerm(this.type, this.logic, this.inputs, this.row, this.col) {
+    delay = 0.0;
+    final deltaDelay = switch (type) {
+      _CompressTermType.carry => carryDelay,
+      _CompressTermType.sum => sumDelay,
+      _CompressTermType.pp => 0.0
+    };
+    for (final i in inputs) {
+      if (i.delay + deltaDelay > delay) {
+        delay = i.delay + deltaDelay;
+      }
+    }
+  }
+  @override
+  int compareTo(Object other) {
+    if (other is! _CompressTerm) {
+      throw Exception('Input must be of type CompressTerm ');
+    }
+    return delay > other.delay ? 1 : (delay < other.delay ? -1 : 0);
+  }
+
+  /// Evaluate the logic value of a given CompressTerm.
+  LogicValue evaluate() {
+    late LogicValue value;
+    switch (type) {
+      case _CompressTermType.pp:
+        value = logic.value;
+      case _CompressTermType.sum:
+        // xor the eval of the terms
+        final termValues = [for (final term in inputs) term.evaluate()];
+        final sum = termValues.swizzle().xor();
+        value = sum;
+      case _CompressTermType.carry:
+        final termValues = [for (final term in inputs) term.evaluate()];
+        final termValuesInt = [
+          for (var i = 0; i < termValues.length; i++) termValues[i].toInt()
+        ];
+
+        final count = (termValuesInt.isNotEmpty)
+            ? termValuesInt.reduce((c, term) => c + term)
+            : 0;
+        final majority =
+            (count > termValues.length ~/ 2 ? LogicValue.one : LogicValue.zero);
+        value = majority;
+    }
+    return value;
+  }
+
+  @override
+  String toString() {
+    final str = StringBuffer();
+    final ts = switch (type) {
+      _CompressTermType.pp => 'pp',
+      _CompressTermType.carry => 'c',
+      _CompressTermType.sum => 's'
+    };
+    str
+      ..write(ts)
+      ..write('$row,$col');
+    return str.toString();
+  }
+}
+
+/// A column of partial product terms
+typedef ColumnQueue = PriorityQueue<_CompressTerm>;
```

Comment:
Should this be private? `_ColumnQueue`?

Reply:
I think I need to factor out a module out of this to create a real API which is in: partialproduct out: LogicArray [2].
During testing, I didn't do this.  This will take me a bit to refactor (maybe not too bad).

Reply:
I've done the refactoring to hide ColumnCompression classes and wrap with an AddendCompression class.